docusaurus/docs/shared/implement-resource-handler.md

```diff
@@ -0,0 +1,189 @@
+
+There are two ways you can implement this in your plugin, [using the `httpadapter` package](#using-the-httpadapter-package) or [manually implementing it](#manually-implementing-backendcallresourcehandler) in your plugin.
+
+### Using the `httpadapter` package
+
+The [`httpadapter`](https://pkg.go.dev/github.com/grafana/grafana-plugin-sdk-go/backend/resource/httpadapter) package provided by the [Grafana Plugin SDK for Go](../../key-concepts/backend-plugins/grafana-plugin-sdk-for-go) is the recommended way for handling resources. This package provides support for handling resource calls using using the [`http.Handler`](https://pkg.go.dev/net/http#Handler) interface and allows responding to HTTP requests in a more Go-agnostic way and makes it easier to support multiple routes and methods (GET, POST etc).
+
+Using [`http.Handler`](https://pkg.go.dev/net/http#Handler) allows you to also use Go’s built-in router functionality called [`ServeMux`](https://pkg.go.dev/net/http#ServeMux) or your preferred HTTP router library (for example, [`gorilla/mux`](https://github.com/gorilla/mux)).
+
+:::note
+
+Go 1.22 [includes routing enhancement](https://go.dev/blog/routing-enhancements) that adds support for method matching and wildcards using the [`ServeMux`](https://pkg.go.dev/net/http#ServeMux).
+
+:::
+
+In the following example we demonstrate using the `httpadapter` package, `ServeMux` and `http.Handler` to add support for retrieving namespaces (`/namespaces`), projects (`/projects`) and updating the state of some device (`/device`) :
+
+```go
+package myplugin
+
+import (
+  "context"
+  "net/http"
+
+  "github.com/grafana/grafana-plugin-sdk-go/backend"
+  "github.com/grafana/grafana-plugin-sdk-go/backend/resource/httpadapter"
+)
+
+type MyPlugin struct {
+  resourceHandler backend.CallResourceHandler
+}
+
+func New() *MyPlugin {
+  p := &MyPlugin{}
+  mux := http.NewServeMux()
+  mux.HandleFunc("/namespaces", p.handleNamespaces)
+  mux.HandleFunc("/projects", p.handleProjects)
+  mux.HandleFunc("/device", p.updateDevice)
+  p.resourceHandler := httpadapter.New(mux)
+  return p
+}
+
+func (p *MyPlugin) CallResource(ctx context.Context, req *backend.CallResourceRequest, sender backend.CallResourceResponseSender) error {
+  return p.resourceHandler.CallResource(ctx, req)
+}
+
+func (p *MyPlugin) handleNamespaces(rw http.ResponseWriter, req *http.Request) {
+  if req.Method != http.MethodGet {
+    rw.WriteHeader(http.StatusNotFound)
+    return
+  }
+
+  ctxLogger := backend.Logger.FromContext(req.Context())
+
+  rw.Header().Add("Content-Type", "application/json")
+  _, err := rw.Write([]byte(`{ "namespaces": ["ns-1", "ns-2"] }`))
+  if err != nil {
+    ctxLogger.Error("Failed to write response", "error", err)
+    return
+  }
+  rw.WriteHeader(http.StatusOK)
+}
+
+func (p *MyPlugin) handleProjects(rw http.ResponseWriter, req *http.Request) {
```

Comment:
IMO there's no need to provide fully functional code snippets for these resource handlers. create-plugin already generates functional code that devs can see working in their code. 

perhaps you can reduce this to simply the response removing all the code in the middle? it can become distracting from the important part which is the resource handler part.

Reply:
Much better. thanks <3